File: src/main/kotlin/pl/taurus/reviseo/persona/application/domain/model/Persona.kt
package pl.taurus.reviseo.persona.application.domain.model

data class Persona(
    val identifier: PersonaIdentifier,
    val name: PersonaName,
    val description: PersonaDescription,
    val customInstructions: CustomInstructions,
    val checklist: Checklist,
    val keyAspects: KeyAspects,
)

File: frontend/src/types/persona.ts
export interface Persona {
  identifier: string;
  name: string;
  description: string;
  customInstructions: string;
  checklist: string[];
  keyAspects: string[];
}

export interface CreatePersonaRequest {
  name: string;
  description: string;
  customInstructions: string;
  checklist: string[];
  keyAspects: string[];
}

export interface UpdatePersonaRequest {
  name: string;
  description: string;
  customInstructions: string;
  checklist: string[];
  keyAspects: string[];
}

export interface MarketplacePersona extends Persona {
  isInstalled: boolean;
}
